### ISSUE TITLE
Panic when running CPU Manager node e2e test on Ubuntu 16.04 locally: "assignment to entry in nil map"

### ISSUE THREAD

**Issue by ScorpioCPH** [kubernetes/kubernetes#56144]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

Test failed when running CPU Manager node e2e test on Ubuntu 16.04 locally, and show this panic message:

```shell
Test Panicked
assignment to entry in nil map
...
.../test/e2e_node/cpu_manager_test.go:160
```
**What you expected to happen**:

Tests passed.

**How to reproduce it (as minimally and precisely as possible)**:

Run this node e2e test, use the following command:

```shell
make test-e2e-node TEST_ARGS='--feature-gates=DynamicKubeletConfig=true' FOCUS="CPU Manager" SKIP="" PARALLELISM=1
```

**Anything else we need to know?**:

I think this is because the following codes:

```go
if _, ok := newCfg.KubeReserved["cpu"]; !ok {
	newCfg.KubeReserved["cpu"] = "200m"
}
```
`newCfg.KubeReserved` is `nil`.

I will send a PR to fix this after it is confirmed.

/sig node
/assign @balajismaniam @ConnorDoyle @derekwaynecarr PTAL, Thanks!

**Comment by ConnorDoyle** [comment]
Thanks for the bug report. Reproduced successfully. It looks like the response body may have changed from empty map to null, which caused this to break. In any case, @balajismaniam suggested adding the fix to his pending PR here: #56105

**Comment by ScorpioCPH** [comment]
@ConnorDoyle Hi, #56105 have fixed this, lgtm.
